internal/highlight/highlight.go

```diff
@@ -238,9 +242,11 @@ var requestCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
 	Help: "Counts syntax highlighting requests and their success vs. failure rate.",
 }, []string{"status"})
 
-func init() {
-	prometheus.MustRegister(requestCounter)
-}
+var requestHistogram = prometheus.NewHistogram(
+	prometheus.HistogramOpts{
+		Name: "src_syntax_highlighting_duration",
```

Comment:
Prometheus guide suggests including the unit in the metric name. This is useful since you don't have to assume it is seconds. https://prometheus.io/docs/practices/naming/

```suggestion
		Name: "src_syntax_highlighting_duration_seconds",
```